ports/stm/supervisor/port.c

```diff
@@ -454,3 +454,18 @@ void _init(void)
 
 }
 
+void HAL_Delay(uint32_t delay) {
+    if (delay > 1000) {
+        mp_hal_delay_ms(delay / 1000);
+        delay %= 1000;
+    }
+    mp_hal_delay_us(delay);
+}
+
+#if 0
+uint32_t HAL_GetTick() {
+    uint8_t subticks;
+    uint32_t result = (uint32_t)port_get_raw_ticks(&subticks);
+    return (result * 1000) + (subticks * 1000 / 32);
+}
+#endif
```

Comment:
This needs review from @hierophect  -- the implementation of HAL_GetTick didn't work, but I believe something's needed